pkg/porter/version.go

```diff
@@ -16,3 +26,55 @@ func (p *Porter) PrintVersion(opts version.Options) error {
 	}
 	return version.PrintVersion(p.Context, opts, metadata)
 }
+
+func getSystemInfo() *SystemInfo {
+	return &SystemInfo{
+		OS:            runtime.GOOS,
+		Arch:          runtime.GOARCH,
+	}
+}
+
+func (system *SystemInfo) printSystemInfo(out io.Writer) error {
+	_, err := fmt.Fprintf(out, "os: %s\n", system.OS)
+	_, err = fmt.Fprintf(out, "arch: %s\n", system.Arch)
+	return err
+}
+
+func printSectionHeader(out io.Writer, header string){
+	underline := "-------"
+	_, _ = fmt.Fprintf(out, "\n%s\n%s\n", header, underline)
+
+}
+
+func (p *Porter) PrintDebugInfo(opts version.Options) error {
+	// force opts to print version as plaintext
```

Comment:
Apologies for not catching this when I first checked the PR, but this new `--system` flag needs to work in conjunction with the `--output plaintext|json` flag. So `PrintDebugInfo` needs to support printing to json as well.